gitlab_test.go

```diff
@@ -0,0 +1,268 @@
+package reviewdog
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/kylelemons/godebug/pretty"
+	"github.com/xanzy/go-gitlab"
+)
+
+const noGitlabTokenSkipTestMes = "skipping test (requires actual Personal access tokens. export REVIEWDOG_TEST_GITLAB_API_TOKEN=<GitLab Personal Access Token>)"
+
+func setupGitLabClient() *gitlab.Client {
+	token := os.Getenv("REVIEWDOG_TEST_GITLAB_API_TOKEN")
+	if token == "" {
+		return nil
+	}
+	cli := gitlab.NewClient(nil, token)
+	cli.SetBaseURL("https://gitlab.com/api/v4")
+	return cli
+}
+
+func TestGitLabMergeRequest_Post(t *testing.T) {
+	t.Skip("skipping test which post comments actually")
+	client := setupGitLabClient()
+	if client == nil {
+		t.Skip(noGitlabTokenSkipTestMes)
+	}
+
+	// https://gitlab.com/nakatanakatana/reviewdog/merge_requests/1
+	owner := "nakatanakatana"
+	repo := "reviewdog"
+	pr := 1
+	sha := "bc328521a974c23acb24e8ebf51c1c2dcdb4fe6a"
+
+	g, err := NewGitLabMergeReqest(client, owner, repo, pr, sha)
+	if err != nil {
+		t.Fatal(err)
+	}
+	comment := &Comment{
+		CheckResult: &CheckResult{
+			Path: "diff.go",
+			Lnum: 22,
+		},
+		LnumDiff: 11,
+		Body:     "[reviewdog] test",
+	}
+	// https://gitlab.com/nakatanakatana/reviewdog/merge_requests/1
```

Comment:
I don't have a permission to this repository.
Can you remove the test which actually sends a request to GitLab?

Reply:
Removed 3 tests.
How about to give repo info as environment variables?
(I may send another pull request)

Reply:
Sounds good.
But the tests assume the content of the merge-request, so I'll first create hard-coded tests.

I can merge p-r to make it env var too.
Also, it's totally optional but if you want to run tests by yourself with the repository I'll create, I'll give you a permission.